contracts/common/AHasForwarder.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "./lib/LibHasForwarder.sol";
+
+/**
+ * @title The Forwarder behaviour abstract contract.
+ * @notice The AHasForwarder abstract contract is in charge of adding the Forwarder
+ *         functionality to any contract inheriting from it.
+ */
+abstract contract AHasForwarder {
+  /// Errors.
+
+  /**
+   * @notice Happens when a function is called by a non forwarder manager.
+   * @param who is the address that called the function.
+   */
+  error RequiresForwarderManager(address who);
+
+  /// Events.
+
+  /**
+   * @notice Emited when a forwarder is set on an implementing contract.
+   * @param forwarderAddress is the address of the trusted forwarder.
+   */
+  event ForwarderChanged(address forwarderAddress);
+
+  // SEE: https://github.com/opengsn/gsn/blob/v3.0.0-beta.10/packages/contracts/src/ERC2771Recipient.sol
+
+  /**
+   * @notice ERC2771Recipient implementation.
+   * @param _forwarderAddress the forwarder address.
+   * @return bool if the forwarder is trusted.
+   */
+  function isTrustedForwarder(address _forwarderAddress) public view returns (bool) {
+    return address(_forwarderAddress) == LibHasForwarder.data().forwarderAddress;
+  }
+
+  /**
+   * @notice ERC2771Recipient implementation.
+   * @param _forwarderAddress the forwarder address.
+   */
+  function setTrustedForwarder(address _forwarderAddress) external onlyForwarderManager {
+    LibHasForwarder.Data storage ds = LibHasForwarder.data();
+    ds.forwarderAddress = _forwarderAddress;
```

Comment:
Are there any tests we want to do before setting a new forwarder? For example check that it implements a particular interface? :)

Reply:
Yep, would be good to handle that 👍

Reply:
Handled here now: https://github.com/tableturn/tt-white-contracts/pull/89/files#diff-3c1a1e97cdd80a3e276c77db6897800d36bd2ff3680a76efb51798c59f0d1d8cR48